### PR TITLE
Test header elements with cypress

### PR DESCRIPTION
**Pull request by gevos**
# How to

Cypress test for header should successfully run

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] Labels applied. if it's a release? a hotfix?
- [x] Tests added.

### CODE REVIEW

File: cypress/integration/Header.spec.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        // Do not test click because of CORS problem
+        // when redirecting to another page
+      });
+
+      it(`should display language select`, () => {
+        setViewPort(viewport);
+        cy.get('.wkp-single-value-wrapper').should('be.visible');
+
+        // Click the react-select element
+        cy.get('.wkp-header .css-m869aq-ValueContainer').click();
```
Comment (edellucien) [inline cypress/integration/Header.spec.js:80]: Isn't this class name gonna change on build? Maybe with an id we could select the correct item, else it is good enough.

File: cypress/integration/Header.spec.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        cy.get('.wkp-topics-menu-body').should('not.be.visible');
+        cy.get('.wkp-menu-header').click();
+
+        // Menu should open
+        cy.get('.wkp-topics-menu-body').should('be.visible');
+      });
+
+      it(`should display WKP login`, () => {
+        setViewPort(viewport);
+        cy.get('.wkp-login').should('be.visible');
```
Comment (edellucien) [inline cypress/integration/Header.spec.js:70]: We should either find a way of testing the redirect or simply not write the test at all I think.

Comment (gevos) [reply cypress/integration/Header.spec.js:70]: I have checked this. These [code samples](https://github.com/geops/trafimage-maps/blob/master/src/components/Login/Login.js#L19) show solutions how to test redirects. As the Login component does not have a href attribute and opens the new page by using window.location.href (see [line](https://github.com/geops/trafimage-maps/blob/master/src/components/Login/Login.js#L19)), we could only use [third option](https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/testing-dom__tab-handling-links/cypress/integration/tab_handling_anchor_links_spec.js#L183) which sets chromeWebSecurity to false. However, when I do this, I get redirected to chrome-error://chromewebdata/. I could try to solve this. But, I think, the main idea of cypress is to test a single domain. 

I would prefer to only check if this Login button is available. It's better than checking nothing.

File: cypress/integration/Header.spec.js (hunk 1 above)
Comment (gevos) [reply cypress/integration/Header.spec.js:80]: @edellucien has found a workaround for the IABP project where a ValueContainer and Menu is defined for the react-select component, additionally to the SingleContainer. So, we have  beautiful css names. Unfortunately, this does not work in trafimage-maps. And we don't know why. We get a detached-from-DOM error. And even, if we do not change the classes in the test, it does not work anymore.

I have checked if the strange react-select css classes change after a built or when the react-select version changes - no. So, I prefer to leave this test the way it is. @edellucien agrees.

**Comment by edellucien** [comment]
Thanks a lot for the tests! :clap:

**Pull request merged**